.github/workflows/publish.yml

```
@@ -0,0 +1,36 @@
+name: Build
+
+on:
+  push:
+    branches: [main]
+    paths: [package.json]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: lts/*
+
+      - name: Get the Yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(corepack yarn config get cacheFolder)"
+
+      - uses: actions/cache@v3
+        with:
+          path: ${{steps.yarn-cache-dir-path.outputs.dir}}
+          key: ${{runner.os}}-yarn-${{hashFiles('**/yarn.lock')}}
+          restore-keys: |
+            ${{runner.os}}-yarn-
+
+      - name: Publish the package
+        run:
+          corepack yarn install --immutable
+          corepack yarn npm publish --tolerate-republish
+        env:
+          YARN_NPM_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
```

Let's create a GitHub release as well. This will also automatically create a git tag for the release.
```suggestion
          YARN_NPM_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
      - name: Get version number
        id: versionNumber
        run: jq -r '"##[set-output name=version;]"+.version' < package.json
      - name: Create GitHub release
        run: gh release create v${{ steps.versionNumber.outputs.version }} -t "Corepack ${{ steps.versionNumber.outputs.version }}"  --prerelease
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

---

Hum but we don't want to create a release if it turns out the commit is not a release actually. Let me check what happens if one tries to release a version that's already been released.

---

The `--tolerate-republish` makes Yarn not do anything, although if we also create a GH release then it won't be enough

---

If we use release-please, that's being taken care of for us, we don't have to bother.